assets/config.php

```diff
@@ -35,6 +35,37 @@
         // Stancl\Tenancy\Bootstrappers\RedisTenancyBootstrapper::class, // Note: phpredis is needed
     ],
 
+
+    /**
+     * Pending tenancy config.
+     * This is useful if you're looking for a way to always have a tenant ready to be used.
+     */
+    'pending' => [
+        /**
+         * If disabled, pending tenants will be excluded from all tenant queries. Unless if
+         * told otherwise with ::withPending() or ::onlyPending().
+         * Note: when disabled, this will also ignore tenants when runnings any tenants commands (migration, seed, etc.)
+         */
+        'include_in_queries' => true,
+        /**
+         * Defines how many tenants you want to be in a pending state.
+         * This value should be changed depending on how often a new tenant is created
+         * and how often you run the `tenancy:pending` command via the scheduler.
```

Comment:
Not sure about this sentence. If it's meant to advise the user about setting the value, it should just say "This depends on the volume of tenants you're creating" or something like that. The "prepared pool" (perhaps we want to use some term like that, I like `pool`) depends on how often you're creating tenants.

Reply:
Right, maybe even the key name could be something like 'pending.pool_size' instead of 'pending.count'

Reply:
Also the CreatePendingTenants command's signature is `tenants:pending`, I don't think that captures what the command does. The signature could be better (e.g. `tenants:create-pending`)

Reply:
"Pending tenancy" is used a lot here (like "Pending tenancy config"), and I'm not really sure about that. "Pending tenant" ("Pending tenant config" etc.) seems more appropriate. What do you think @stancl?

Reply:
Updated the comments, and the naming of some variables, you can check out the changes [here](https://github.com/archtechx/tenancy/compare/4d7831d6f6a172760dd54f64e527ea9895768e02..stein-j-readied-tenant)

Reply:
> pending.pool_size

Note that I'm not saying we should just stick the word `pool` here, we should use consistent terminology everywhere. I'm just saying that if we need a term to refer to this "pool of prepared tenants", `pool` *could* work well.

But the used terminology is up to you — at this point. I'll do a review when you fix all of the writing and then we can change the terminology if needed. Right now it should just be descriptive and consistent.

> The signature could be better

Agree.

> Pending tenancy

I don't like this term, it isn't very self-explanatory. "Pending tenants" is fine, pending tenancy is likely not what we want.

Reply:
I think the writing is fine now. But I don't like [this config section](https://github.com/archtechx/tenancy/pull/869/files#diff-579787e0054e183a7e66b899216a02aef9cec3bdf11c4c298f052a14c0699b03R56-R64). Since the time constraints are used only in the ClearPendingTenants command, I think it would be better to just schedule the command with the time constraints as options instead of having them in the config.

Reply:
> I think it would be better to just schedule the command with the time constraints as options instead of having them in the config

Hmm that could be a solution, yeah. I do like having all of the config in one place, but if it's making the config file significantly noisier and is only being used by the command, then we might move it there.

Two questions:
- Does the user have to schedule the command manually now?
- Can you provide an example for how the command would be scheduled with these options?

Reply:
> Does the user have to schedule the command manually now?

Yeah. There was a comment saying "For this to work automatically, make sure to call the `tenancy:pending-clear` command in the scheduler." (changed it later so it's more concise, but that doesn't matter)

> Can you provide an example for how the command would be scheduled with these options?

`$schedule->command('tenants:pending-clear --older-days=5')`

or, if you'd set `older_than_days` to 5 in the config (the config value gets used if the option is null):

`$schedule->command('tenants:pending-clear)`

Reply:
In that case yeah, we can make it an argument. I'd consider having it in the config if the command were scheduled automatically (which comes with a separate deal of complexity), but if the user is scheduling the command **anyway**, we can just specify those values as arguments.